src/store/keycodes.js

```diff
@@ -0,0 +1,218 @@
+    if (!isUndefined(code)) {
+      if (
+        code.includes(filter) ||
+        name?.toUpperCase().includes(filter) ||
+        title?.toUpperCase().includes(filter)
+      ) {
+        matchCount += 1;
+      }
+    }
+    return matchCount;
+  }, 0);
+}
+
+/**
+ * Use Options style for now.
+ */
+export const useKeycodesStore = defineStore('keycodes', {
+  /**
+   *
+   * @returns {KeycodeStoreState}
+   */
+  state: () => ({
+    keycodes: [
+      ...keycodePickerTabLayout.ANSI_ISO,
+      ...keycodePickerTabLayout.special
+    ],
+    searchFilter: '',
+    searchCounters: {
+      ANSI: 0,
+      'ISO/JIS': 0,
+      Quantum: 0,
+      KeyboardSettings: 0,
+      AppMediaMouse: 0
+    },
+    steno: false,
+    active: 'ANSI'
+  }),
+  getters: {
+    lookupKeyPressCode: () => (searchTerm) =>
+      this.lookupKeycode(searchTerm, true),
+    lookupKeycode:
+      (state) =>
+      (searchTerm, isKeys = false) =>
+        state.keycodes.find(
+          ({ code, keys }) =>
+            code === searchTerm || (isKeys && keys && keys === searchTerm)
+        )
+  },
+  actions: {
+    changeActive(newActive) {
+      this.active = newActive;
+    },
+    enableSteno() {
+      this.steno = true;
+      this.keycodes = generateKeycodes(getOSKeyboardLayout(), this.steno);
+    },
+    disableSteno() {
+      this.steno = false;
+      this.keycodes = generateKeycodes(getOSKeyboardLayout(), this.steno);
+    },
+    updateKeycodeNames() {
+      this.keycodes = generateKeycodes(getOSKeyboardLayout(), this.steno);
+    },
+    setSearchFilter(newVal) {
+      this.searchFilter = newVal;
+      if (this.searchFilter !== '') {
+        this.searchCounters = {
+          ANSI: countMatches(this.searchFilter, ansi),
+          'ISO/JIS': countMatches(this.searchFilter, iso_jis),
+          Quantum: countMatches(this.searchFilter, quantum),
+          KeyboardSettings: countMatches(this.searchFilter, settings),
+          AppMediaMouse: countMatches(this.searchFilter, media)
+        };
+      }
+    }
+  }
+});
+
+/**
+ * @typedef {Object} KeycodeLabel - used to label groups of keycodes
+ * @property {string} label - Label name
+ * @property {'label'} width - always the special indicator 'label'
+ * @property {boolean} [group] - group following keycodes under this
+ * @property {string} [icon] - font-awesome icon to display
+ * @property {string} [iconClass] - css class to apply
+ */
+
+/**
+ * @typedef {Object} WidthPlaceholder - used for spacing
+ * @property {number} width - width in Key Units * 1000. e.g. 1U = 1000, 2U = 2000.
+ */
+
+/**
+ * @typedef {Object} KeycodeDefinition - metadata about a keycode
+ * @property {string} name - UI display label for the keycode
+ * @property {string} code - QMK keycode defintion
+ * @property {string} [keys] - javascript keypress id. Used by keyboard handler
+ * @property {number} [width] - width in Key Units * 1000. e.g. 1U = 1000, 2U = 2000.
```

Comment:
```suggestion
 * @typedef {Object} WidthPlaceholder - used for spacing
 * @property {number} width - width in Key Units * 1000. e.g. 1U = 1000, 2U = 2000
 */

/**
 * @typedef {Object} KeycodeDefinition - metadata about a keycode
 * @property {string} name - UI display label for the keycode
 * @property {string} code - QMK keycode definition
 * @property {string} [keys] - javascript keypress id. Used by keyboard handler
 * @property {number} [width] - width in Key Units * 1000. e.g. 1U = 1000, 2U = 2000
 ```
 Typo + inconsistent use of trailing period